R/tt_afun_utils.R

```diff
@@ -278,44 +265,32 @@ in_rows <- function(..., .list = NULL, .names = NULL,
-#'   NULL-ed by the returned analysis function. Defaults to \code{TRUE} if
-#'   \code{fun} accepts \code{.in_ref_col} as a formal argument. Note this
-#'   argument occurs after \code{...} so it must be \emph{fully} specified  by
-#'   name when set.
-#' @param .format_na_strs ANY. vector/list of \code{na} strings to override any
-#'   defaults applied by \code{fun}.
-#' @return A function suitable for use in \code{\link{analyze}} with element
-#'   selection, reformatting, and relabeling performed automatically.
-#'
-#' @note setting \code{.ungroup_stats} to non-null changes the \emph{structure}
-#'   of the  value(s) returned by \code{fun}, rather than just labeling
-#'   (\code{.labels}), formatting (\code{.formats}), and selecting amongst
-#'   (\code{.stats}) them. This means that subsequent \code{make_afun} calls to
-#'   customize the output further both can and must operate on the new
-#'   structure, \emph{NOT} the original structure returned by \code{fun}. See
-#'   the final pair of examples below.
+#' Create a custom analysis function wrapping an existing function
+#'
+#' @param fun (`function`)\cr the function to be wrapped in a new customized analysis function.
+#'   `fun` should return a named `list`.
+#' @param .stats (`character`)\cr names of elements to keep from `fun`'s full output.
+#' @param .formats (`ANY`)\cr vector or list of formats to override any defaults applied by `fun`.
+#' @param .labels (`character`)\cr vector of labels to override defaults returned by `fun`.
+#' @param .indent_mods (`integer`)\cr named vector of indent modifiers for the generated rows.
+#' @param .ungroup_stats (`character`)\cr vector of names, which must match elements of `.stats`.
+#' @param ... additional arguments to `fun` which effectively become new defaults. These can still be
+#'   overridden by `extra_args` within a split.
+#' @param .null_ref_cells (`flag`)\cr whether cells for the reference column should be `NULL`-ed by the
+#'   returned analysis function. Defaults to `TRUE` if `fun` accepts `.in_ref_col` as a formal argument. Note
+#'   this argument occurs after `...` so it must be *fully* specified by name when set.
+#' @param .format_na_strs (`ANY`)\cr vector/list of `NA` strings to override any defaults applied by `fun`.
+#'
+#' @return A function suitable for use in [analyze()] with element selection, reformatting, and relabeling
+#'   performed automatically.
 #'
-#' @seealso [analyze()]
+#' @note
+#' Setting `.ungroup_stats` to non-`NULL` changes the *structure* of the value(s) returned by `fun`, rather than
+#' just labeling (`.labels`), formatting (`.formats`), and selecting amongst (`.stats`) them. This means that
+#' subsequent `make_afun` calls to customize the output further both can and must operate on the new structure,
+#' *not* the original structure returned by `fun`. See the final pair of examples below.
 #'
-#' @export
+#' @seealso [analyze()]
```

Comment:
is there also a link to this from afun? or from analyze seealso? or both?

Reply:
It was my understanding that this function was mainly used in `tern` with the idea that it could possibly be phased out at some point, since there are some very `tern`-specific parts of the function, and this is why it is not very visible to users.

Reply:
It is clear now. Thanks Emily!